advanced_alchemy/base.py

```diff
@@ -272,13 +276,13 @@ def __table_args__(cls) -> tuple | dict:
 
 
 def create_registry(
-    custom_annotation_map: dict[type, type[TypeEngine[Any]] | TypeEngine[Any]] | None = None,
+    custom_annotation_map: dict[Any, type[TypeEngine[Any]] | TypeEngine[Any]] | None = None,
```

Comment:
SQLAlchemy type this as `Any` also: https://github.com/sqlalchemy/sqlalchemy/blob/e69ca167fa0dcd33ef8c7dccbccf127c8665472e/lib/sqlalchemy/orm/decl_api.py#L101